cumulusci/tasks/github/commit_status.py

```diff
@@ -19,6 +19,12 @@ class GetPackageDataFromCommitStatus(BaseGithubTask, BaseSalesforceApiTask):
         "version_id": {"description": "Package version id"},
     }
 
+    def _init_api(self, base_url=None):
+        if float(self.project_config.project__package__api_version) < 44.0:
+            # We need at least API 44.0 to query for the Dependencies field.
+            self.api_version = "44.0"
+        return super()._init_api(base_url=base_url)
```

Comment:
This task only does the one query against the org, so I think we can simply set `api_version` to a specific version that we know works, and not worry about overriding the project api_version.